### PR TITLE
MAT-10003: Implement VSAC value set expansion fetch when the target value set was not found in TxFHIR.

### PR DESCRIPTION
**Pull request by jkotanchik-SB**
## Terminology Service PR

Jira Ticket: [MAT-10003](https://jira.cms.gov/browse/MAT-10003)
(Optional) Related Tickets:

### Summary

Implement secondary/backup flow.  

When a Value Set expansion is unavailable in TxFHIR, attempt to retrieve from VSAC. Handle pagination on VSAC results when necessary.

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/terminology/service/ValueSetExpansionService.java [hunk 1]
```diff
@@ -178,15 +180,33 @@ private Optional<ValueSet> fetchExpansionFromTxFhir(MadieValueSet madieValueSet)
...
-    // not their OIDs. This necessitates using the URL query-param when requesting
-    // expansions from VSAC, which differs from our current flow.
+        url,
+        version);
 
-    return Optional.empty();
+    String urlWithVersion =
+        StringUtils.isNotBlank(version) ? url + "&valueSetVersion=" + version : url;
+    String urlWithOffset = offset > 0 ? urlWithVersion + "&offset=" + offset : urlWithVersion;
+    String expandUrl = "https://cts.nlm.nih.gov/fhir/ValueSet/$expand?url=" + urlWithOffset;
```
Comment (chubert-sb) [inline src/main/java/gov/cms/madie/terminology/service/ValueSetExpansionService.java]: This would probably be good to set as a variable

**Review by chubert-sb — DISMISSED** [review]

File: src/main/java/gov/cms/madie/terminology/service/ValueSetExpansionService.java (hunk 1 above)
Comment (jkotanchik-SB) [reply src/main/java/gov/cms/madie/terminology/service/ValueSetExpansionService.java]: True. Done.

**Review by chubert-sb — APPROVED** [review]
